test/validation/api/classification/odp_classification_tests.c

```diff
@@ -683,35 +691,60 @@ void test_pktio_pmr_composite_cos(void)
 
 static void classification_test_pktio_configure(void)
 {
+	odp_cls_capability_t capa;
+	int num_cos;
+
+	odp_cls_capability(&capa);
+	num_cos = capa.max_cos;
+
 	/* Configure the Different CoS for the pktio interface */
-	if (TEST_DEFAULT)
+	if (num_cos >= NUM_COS_DEFAULT && TEST_DEFAULT) {
 		configure_pktio_default_cos();
-	if (TEST_ERROR)
+		tc.default_cos = 1;
+		num_cos -= NUM_COS_DEFAULT;
+	}
+	if (num_cos >= NUM_COS_ERROR && TEST_ERROR) {
 		configure_pktio_error_cos();
-	if (TEST_PMR_CHAIN)
+		tc.error_cos = 1;
+		num_cos -= NUM_COS_ERROR;
+	}
+	if (num_cos >= NUM_COS_PMR_CHAIN && TEST_PMR_CHAIN) {
 		configure_cls_pmr_chain();
-	if (TEST_L2_QOS)
+		tc.pmr_chain = 1;
+		num_cos -= NUM_COS_PMR_CHAIN;
+	}
+	if (num_cos >= NUM_COS_L2_PRIO && TEST_L2_QOS) {
 		configure_cos_with_l2_priority();
-	if (TEST_PMR)
+		tc.l2_priority = 1;
+		num_cos -= NUM_COS_L2_PRIO;
+	}
+	if (num_cos >= NUM_COS_PMR && TEST_PMR) {
 		configure_pmr_cos();
-	if (TEST_PMR_SET)
+		tc.pmr_cos = 1;
+		num_cos -= NUM_COS_PMR;
+	}
+	if (num_cos >= NUM_COS_COMPOSITE && TEST_PMR_SET) {
 		configure_pktio_pmr_composite();
+		tc.pmr_composite_cos = 1;
+		num_cos -= NUM_COS_COMPOSITE;
+	}
+
 }
 
 static void classification_test_pktio_test(void)
 {
 	/* Test Different CoS on the pktio interface */
-	if (TEST_DEFAULT)
+	if (tc.default_cos && TEST_DEFAULT)
```

Comment:
Since the various `tc` bits are only set if the corresponding `#define TEST_xxx` is set, the `&& TEST_xxx` part of these conditionals is redundant.